### DISCUSSION TITLE
Self hosting ISR Different Back-ends

### DISCUSSION THREAD

**Discussion by razilevin** [vercel/next.js#45324] [Help] [answered]
Is there an extension mechanism where we can plugin our own custom ISR data-store which is not in memory or a file-system. E.g. back in the days we used to write web-pages to redis with the key being the path of the page. Then serving those requests from redis if there and if not passing that request along to be generated. 

This way we do not need to mount a shared file-system in k8s or the memory store where each instance needs to get a request to render the content.

**Comment by leerob** [comment, marked as answer]
Hey – we have just published new documentation on self-hosting, including how to configure a custom cache handler for use with ISR / Data Cache in both the Pages and App Router. These docs go in depth about how other features like Image Optimization, Middleware, Environment Variables, Build Caching, and more work when self-hosted. There's also updated Docker based examples.

https://nextjs.org/docs/app/building-your-application/deploying
